fhir-search/src/test/java/com/ibm/fhir/search/parameters/ParametersSearchUtilTest.java

```diff
@@ -146,11 +148,39 @@ public void testGetSearchParameters6Default() throws Exception {
         List<SearchParameter> result = SearchUtil.getApplicableSearchParameters("Patient");
         assertNotNull(result);
         printSearchParameters("testGetSearchParameters6/Patient", result);
-        assertEquals(35, result.size());
+        assertEquals(37, result.size());
 
         result = SearchUtil.getApplicableSearchParameters("Device");
         assertNotNull(result);
         printSearchParameters("testGetSearchParameters6/Device", result);
         assertEquals(18, result.size());
     }
+
+    @Test
+    public void testVersionedSearchParameterFilter() throws Exception {
+        // Test filtering of search parameters for Patient (default tenant).
+        FHIRRequestContext.set(new FHIRRequestContext("tenant4"));
+
+        List<SearchParameter> result = SearchUtil.getApplicableSearchParameters("Device");
+        assertNotNull(result);
+        printSearchParameters("testVersionedSearchParameterFilter/Device", result);
+        for (SearchParameter sp : result) {
+            System.out.println(sp.getUrl().getValue() + "|" + sp.getVersion().getValue());
+            if ("http://hl7.org/fhir/us/core/SearchParameter/us-core-device-type".equals(sp.getUrl().getValue())) {
+                assertNotEquals("4.0.0", sp.getVersion());
+            }
+        }
+
+        FHIRRequestContext.set(new FHIRRequestContext("tenant5"));
+
+        result = SearchUtil.getApplicableSearchParameters("Device");
+        assertNotNull(result);
+        printSearchParameters("testVersionedSearchParameterFilter/Device", result);
+        for (SearchParameter sp : result) {
+            System.out.println(sp.getUrl().getValue() + "|" + sp.getVersion().getValue());
+            if ("http://hl7.org/fhir/us/core/SearchParameter/us-core-device-type".equals(sp.getUrl().getValue())) {
+                assertNotEquals("3.1.1", sp.getVersion());
```

Comment:
small comment on this test and the for loop above. in rare circumstances the for loop may complete without actually hitting an assert condition.

Reply:
good catch.  I updated it to ensure that the expected search parameter is found (or else it fails)